src/service/line.ts

```
@@ -0,0 +1,67 @@
+import { ux } from '@oclif/core';
+import type { ScanResultComponent } from './nes/modules/sbom.ts';
+
+export interface Line {
+  daysEol?: number;
+  purl: ScanResultComponent['purl'];
+  info?: {
+    eolAt?: Date;
+    isEol: boolean;
+  };
+  status: ScanResultComponent['status'];
+}
+
+export function daysBetween(date1: Date, date2: Date) {
+  const msPerDay = 1000 * 60 * 60 * 24 + 15; // milliseconds in a day plus 15 ms
+  return Math.round((date2.getTime() - date1.getTime()) / msPerDay);
+}
+
+export function getMessageAndStatus(status: string, eolAt?: Date) {
+  let msg = '';
+  let stat = '';
+
+  const stringifiedDaysEol = eolAt ? Math.abs(daysBetween(new Date(), eolAt)).toString() : 'unknown';
+
+  switch (status) {
+    case 'EOL': {
+      stat = ux.colorize('red', 'EOL');
+      msg = `EOL'd ${ux.colorize('red', stringifiedDaysEol)} days ago.`;
+      break;
+    }
+
+    case 'LTS': {
+      stat = ux.colorize('yellow', 'LTS');
+      msg = `Will go EOL in ${ux.colorize('yellow', stringifiedDaysEol)} days.`;
+      break;
+    }
+
+    case 'OK': {
+      stat = ux.colorize('green', 'OK');
+      break;
+    }
+    default:
+      throw new Error(`Unknown status: ${status}`);
+  }
+
+  return { stat, msg };
+}
+
+export function formatLine(l: Line, idx: number, ctx: { longest: number; total: number }) {
+  let { info, purl, status } = l;
+
+  info = info || { eolAt: new Date(), isEol: false };
+
+  if (info.isEol && status !== 'EOL') {
```

@dwelch2344  is this correct? This error changes the previous behavior, but it seems like we should throw an error in this case?